pkg/git/repo.go

```diff
@@ -25,7 +25,8 @@ import (
 )
 
 var (
-	ErrNoChange = errors.New("no change")
+	ErrNoChange    = errors.New("no change")
+	ErrForceNeeded = errors.New("some refs were not updated")
```

Comment:
nit: How about `ErrBranchNotFresh`?

Reply:
Nice. applied to https://github.com/pipe-cd/pipecd/pull/3652/commits/74f707015736e6499aedd99d9d417981095bd6e1.